themes/default/content/blog/cloud-engineering-the-future-is-now/index.md

```diff
@@ -0,0 +1,42 @@
+
+Keith Redman, VP of SaaS Engineering at Fernergo, coming from the financial services industry, reiterated the need for a deployment framework that bridged dev and ops teams and an automated build and release process to maintain integrity and reliability. It required a change in culture from teams handing off tasks to each other without input from the other to a completely integrated team where all actions with infrastructure begin with a code commit. The culture shift to an engineering managed process also changed hiring patterns from operators and system administrators with some coding skills to software engineers interested in infrastructure.
+
+In the current cloud engineering landscape, we will continue to see movement towards heterogeneous architectures, an increase in the use of software engineering methodologies to deploy infrastructure, and more automation for building and deploying in the cloud. This segues into the future of cloud engineering with a focus on tooling, process, and teams.
+
+## The Future of Cloud Engineering
+
+The cloud engineering panel included Justin Fitzhugh, Dana Lawson, VP of Engineering Github, Charity Majors, CTO Honeycomb, and Cat Cosgrove, Developer Advocate JFrog, moderating. The session, titled "Your Peers," discussed the future of cloud engineering.
+
+When planning for the future of modern cloud applications, the panel agreed that the industry would continue to iterate and evolve away from DSLs and towards software engineering practices for infrastructure. They also agreed that **go** would be the most popular language for Iinfrastructure as Code, but Ruby and JavaScript will continue to be key languages.
+
+Cloud engineering personas will change as new technologies such as the Internet of Things, availability of 5G, orchestration, edge computing, and future network APIs roll out. Cloud engineers should continue to focus on containers and what goes into applications. Knowing the entire lifecycle of their application from development to production is a must for cloud engineers. It underlies the concept of a single persona that can use services to compose development and services. Ultimately, the job of the cloud engineer is to reduce complexity and reduce friction in the pipeline through using patterns, as needed, and building platforms and tools consistently.
+
+The panel discussed the reality of testing in production. Charity Majors highlighted that you have no choice, whether you admit it or not; there will always be known unknowns because devs or operators cannot test everything. Software will always have bugs, but the problem can be managed by having more ways to separate problems and test them in a small and controlled way. Testing in production begins with observability, starting with usual tools such as strace and other enhanced toolsets to inspect your code. Also, feature flags and deployment models help find bugs and limit their impact when applications go into production.
```

Comment:
```suggestion
The panel discussed the reality of testing in production. Charity Majors highlighted that you have no choice, whether you admit it or not; there will always be known unknowns because devs or operators cannot test everything. Software will always have bugs, but the problem can be managed by having more ways to separate problems and test them in a small and controlled way. Testing in production begins with observability, starting with usual tools such as `strace` and other enhanced toolsets to inspect your code. Also, feature flags and deployment models help find bugs and limit their impact when applications go into production.
```